sycl/test/basic_tests/queue/release.cpp

```diff
@@ -19,3 +19,4 @@ int main() {
 //CHECK: ---> piContextRelease(
 //CHECK: ---> piKernelRelease(
 //CHECK: ---> piProgramRelease(
+//CHECK: ---> piDeviceRelease(
```

Comment:
As far as I understood, the test allowed for the leak. Hence, did it contain some TODO regarding this leak?

Reply:
This test was originally for a different resource leak.  But I can check for the device one easily enough as well by adding this check.

It's hard to make a perfect device check test, because any given testing machine may have a different number of devices that get retained/released.  But at minimum, we used to have no calls to piDeviceRelease, so this ensures that at least something is getting released after the other things. That's reassuring.